eth/client.go

```diff
@@ -155,6 +166,62 @@ func (cl *Client) SubscribeToBlocks() (*BlockSubscription, error) {
 	return &BlockSubscription{sub, blocks, quit}, nil
 }
 
+// SubscribeToBlocksStartingFrom subscribes the client to the mined Ethereum
+// blocks starting from the given block number.
+func (cl *Client) SubscribeToBlocksStartingFrom(startBlockNumber *big.Int) (*BlockSubscription2, error) {
+	headers := make(chan *types.Header)
+	blocks := make(chan *tee.Block)
+	errChan := make(chan error)
+
+	ctx, cancel := NewDefaultContext()
+	defer cancel()
+
+	headerSub, err := cl.SubscribeNewHead(ctx, headers)
+	if err != nil {
+		return nil, fmt.Errorf("subscribing to blockchain head: %w", err)
+	}
+
+	blockSub := &BlockSubscription2{headerSub, blocks, errChan, startBlockNumber}
+
+	errg := errors.NewGatherer()
+	errg.Go(func() error {
```

Comment:
I wouldn't use an error gatherer to start a single routine and then start another routine just to write the result in the channel. you could do here e.g.
```suggestion
	subFn := func() error { // function body as it is }
	go func() {
		errChan <- subFn()
	}()
```

Reply:
should be
```
go func() {
	err := subFn()
	errChan <- err
}()
```

ok, done